lib/credentials/shared_ini_file_credentials.js

```diff
@@ -105,9 +106,16 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
         return;
       }
 
-      this.accessKeyId = profile['aws_access_key_id'];
-      this.secretAccessKey = profile['aws_secret_access_key'];
-      this.sessionToken = profile['aws_session_token'];
+      if (profile['credential_process']) {
```

Comment:
To match [current AWS CLI behavior](https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L86), static credentials should take higher precedence than a `credential_process`.